### DISCUSSION TITLE
Start using ngrx/effects for this

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#287] [Announcements]
# Start using ngrx/effects for this

You're probably only using ngrx/effects to handle the communication to an external source by triggering an effect with a NgRx action. But you can do more.

https://timdeschryver.dev/blog/start-using-ngrx-effects-for-this

**Comment by danvod** [comment]
When using `concatLatestFrom`, is there a good way to make Typescript happy when filtering and using the values that can be null or undefined?

For example, consider this effect 

```
return this.action$.pipe(
	ofType(ExpenseActions.loadExpensesForMonth),
	concatLatestFrom(() => this.store.select(ExpenseActions.selectCurrentPeriod)),
	filter(([_, currentPeriod]) => !!currentPeriod),
	concatMap(([_, currentPeriod]) => of(null) //return not relevant
	),
		
```

In the `concatMap` - TS still thinks currentPeriod is Date | null.

Things I considered so far
1. `filter((x): x is [TypedAction<any>, Date] => !!x[1]),` - technically works, but I don't like the any, and it feels really verbose. This can probably be written more elegantly though.
2. after `concatLatestFrom``, map the tuple into an object and create a custom type guard


How do you approach this, is there something I am missing?
